presto-main-base/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -1479,6 +1489,23 @@ public void addConstraint(Session session, TableHandle tableHandle, TableConstra
         metadata.addConstraint(session.toConnectorSession(connectorId), tableHandle.getConnectorHandle(), tableConstraint);
     }
 
+    @Override
+    public String normalizeIdentifier(Session session, String catalogName, String identifier)
+    {
+        long startTime = System.nanoTime();
+        String normalizedString = identifier.toLowerCase(ENGLISH);
+        Optional<CatalogMetadata> catalogMetadata = getOptionalCatalogMetadata(session, transactionManager, catalogName);
+        if (catalogMetadata.isPresent()) {
+            ConnectorId connectorId = catalogMetadata.get().getConnectorId();
+            ConnectorMetadata metadata = catalogMetadata.get().getMetadataFor(connectorId);
+            normalizedString = metadata.normalizeIdentifier(session.toConnectorSession(connectorId), identifier);
+            session.getRuntimeStats().addMetricValue(GET_IDENTIFIER_NORMALIZATION_TIME_NANOS, NANO, System.nanoTime() - startTime);
+            return normalizedString;
```

Comment:
I still don't see why you need to return here? It looks like you `session.getRuntimeStats().addMetricValue(GET_IDENTIFIER_NORMALIZATION_TIME_NANOS, NANO, System.nanoTime() - startTime);` is the same below, so just assign `normalizedString` and let it return at the end of the method

Reply:
You are right, I overlooked it. We can remove `return normalizedString;` and line before this. I will make the changes. Thanks!